plugins/woocommerce/tests/php/src/Blocks/StoreApi/Utilities/SanitizationUtilTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Automattic\WooCommerce\Tests\Utilities;
+
+use Automattic\WooCommerce\StoreApi\Utilities\SanitizationUtils;
+
+/**
+ * A collection of tests for the array utility class.
+ */
+class SanitizationUtilTest extends \WC_Unit_Test_Case {
+
+
+	/**
+	 * @testdox `kses_array` should return an array of the same shape, but strings should be cleaned with `wp_kses`.
+	 */
+	public function test_kses_array() {
```

Comment:
Once again, looking at the [`wp_kses()`](https://developer.wordpress.org/reference/functions/wp_kses/) source we might want to test sending heavily, beyond any salvation, malformed HTML 👿

Reply:
Feels like that test is more focused on expecting `wp_kses` to behave a certain way no? I've added a test with a more mangled HTML string but I don't know if we should go further than that.